File: examples/flutter/lib/src/models/rule.dart
part of appwrite.models;

/// Rule
class Rule {
    /// Rule ID.
    final String $id;
    /// Rule Collection.
    final String $collection;
    /// Rule type. Possible values: 
    final String type;
    /// Rule key.
    final String key;
    /// Rule label.
    final String label;
    /// Rule default value.
    final String xdefault;
    /// Is array?
    final bool array;
    /// Is required?
    final bool xrequired;
    /// List of allowed values
    final List list;

    Rule({
        required this.$id,
        required this.$collection,
        required this.type,
        required this.key,
        required this.label,
        required this.xdefault,
        required this.array,
        required this.xrequired,
        required this.list,
    });

    factory Rule.fromMap(Map<String, dynamic> map) {
        return Rule(
            $id: map['\$id'],
            $collection: map['\$collection'],
            type: map['type'],
            key: map['key'],
            label: map['label'],
            xdefault: map['default'],
            array: map['array'],
            xrequired: map['required'],
            list: map['list'],
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "\$id": $id,
            "\$collection": $collection,
            "type": type,
            "key": key,
            "label": label,
            "default": xdefault,
            "array": array,
            "required": xrequired,
            "list": list,
        };
    }
}

File: examples/swift-client/Sources/AppwriteModels/Rule.swift

/// Rule
public class Rule {

    /// Rule ID.
    public let id: String

    /// Rule Collection.
    public let collection: String

    /// Rule type. Possible values: 
    public let type: String

    /// Rule key.
    public let key: String

    /// Rule label.
    public let label: String

    /// Rule default value.
    public let xdefault: String

    /// Is array?
    public let array: Bool

    /// Is required?
    public let xrequired: Bool

    /// List of allowed values
    public let list: [Any]

    init(
        id: String,
        collection: String,
        type: String,
        key: String,
        label: String,
        xdefault: String,
        array: Bool,
        xrequired: Bool,
        list: [Any]
    ) {
        self.id = id
        self.collection = collection
        self.type = type
        self.key = key
        self.label = label
        self.xdefault = xdefault
        self.array = array
        self.xrequired = xrequired
        self.list = list
    }

    public static func from(map: [String: Any]) -> Rule {
        return Rule(
            id: map["$id"] as! String,
            collection: map["$collection"] as! String,
            type: map["type"] as! String,
            key: map["key"] as! String,
            label: map["label"] as! String,
            xdefault: map["default"] as! String,
            array: map["array"] as! Bool,
            xrequired: map["required"] as! Bool,
            list: map["list"] as! [Any]
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "$id": id as Any,
            "$collection": collection as Any,
            "type": type as Any,
            "key": key as Any,
            "label": label as Any,
            "xdefault": xdefault as Any,
            "array": array as Any,
            "xrequired": xrequired as Any,
            "list": list as Any
        ]
    }
                                        
}